libs/designer/src/lib/core/actions/bjsworkflow/serializer.ts

```diff
@@ -243,6 +243,20 @@ const getWorkflowParameters = (
   }, {});
 };
 
+export const parseWorkflowParameterValue = (parameterType: any, parameterValue: any) => {
+  const isStringParameter =
+    equals(parameterType, UIConstants.WORKFLOW_PARAMETER_TYPE.STRING) ||
+    equals(parameterType, UIConstants.WORKFLOW_PARAMETER_TYPE.SECURE_STRING);
+
+  return isStringParameter
+    ? parameterValue
+    : parameterValue === ''
+      ? undefined
+      : typeof parameterValue !== 'string'
+        ? parameterValue
+        : JSON.parse(parameterValue);
```

Comment:
I would wrap JSON.parse in a try catch block to not crash if parse fails

Reply:
I was re-using the part of the function from above function, but yes I wrapped it in try catch block and have the above function use this helper now!